exchanges/btcmarkets/btcmarkets.go

```diff
@@ -52,316 +75,472 @@ type BTCMarkets struct {
-	if err != nil {
-		return nil, err
+	if targetAmount > 0 {
+		req["targetAmount"] = strconv.FormatFloat(targetAmount, 'f', -1, 64)
 	}
-
-	if !resp.Success {
-		return nil, errors.New(resp.ErrorMessage)
+	if timeInForce != "" {
+		req["timeInForce"] = timeInForce
 	}
+	req["postOnly"] = postOnly
+	if selfTrade != "" {
+		req["selfTrade"] = selfTrade
+	}
+	if clientOrderID != "" {
+		req["clientOrderID"] = clientOrderID
+	}
+	return resp, b.SendAuthenticatedRequest(http.MethodPost, btcMarketsOrders, req, &resp)
+}
 
-	for i := range resp.Orders {
-		resp.Orders[i].Price /= common.SatoshisPerBTC
-		resp.Orders[i].OpenVolume /= common.SatoshisPerBTC
-		resp.Orders[i].Volume /= common.SatoshisPerBTC
+// GetOrders returns current order information on the exchange
+func (b *BTCMarkets) GetOrders(marketID, limit, status string) ([]OrderData, error) {
+	var resp []OrderData
+	params := url.Values{}
+	params.Set("marketId", marketID)
+	params.Set("limit", limit)
+	params.Set("status", status)
+	return resp, b.SendAuthenticatedRequest(http.MethodGet,
+		common.EncodeURLValues(btcMarketsOrders, params), nil, &resp)
+}
 
-		for x := range resp.Orders[i].Trades {
-			resp.Orders[i].Trades[x].Fee /= common.SatoshisPerBTC
-			resp.Orders[i].Trades[x].Price /= common.SatoshisPerBTC
-			resp.Orders[i].Trades[x].Volume /= common.SatoshisPerBTC
+// CancelOpenOrders cancels all open orders unless pairs are specified
+func (b *BTCMarkets) CancelOpenOrders(marketIDs []string) ([]CancelOrderResp, error) {
+	var resp []CancelOrderResp
+	req := make(map[string]interface{})
+	if len(marketIDs) > 0 {
+		var strTemp string
+		for x := range marketIDs {
+			strTemp += fmt.Sprintf("marketId=%s&", marketIDs[x])
 		}
+		req["marketId"] = strTemp[:len(strTemp)-1]
 	}
-	return resp.Orders, nil
+	return resp, b.SendAuthenticatedRequest(http.MethodDelete, btcMarketsOrders, req, &resp)
 }
 
-// GetAccountBalance returns the full account balance
-func (b *BTCMarkets) GetAccountBalance() ([]AccountBalance, error) {
-	var balance []AccountBalance
+// FetchOrder finds order based on the provided id
+func (b *BTCMarkets) FetchOrder(id string) (OrderData, error) {
+	var resp OrderData
+	return resp, b.SendAuthenticatedRequest(http.MethodGet, btcMarketsOrders+"/"+id,
+		nil, &resp)
+}
 
-	err := b.SendAuthenticatedRequest(http.MethodGet, btcMarketsAccountBalance, nil, &balance)
-	if err != nil {
-		return nil, err
-	}
+// RemoveOrder removes a given order
+func (b *BTCMarkets) RemoveOrder(id string) (CancelOrderResp, error) {
+	var resp CancelOrderResp
+	return resp, b.SendAuthenticatedRequest(http.MethodDelete, btcMarketsOrders+"/"+id,
+		nil, &resp)
+}
 
-	// All values are returned in Satoshis, even for fiat currencies.
-	for i := range balance {
-		balance[i].Balance /= common.SatoshisPerBTC
-		balance[i].PendingFunds /= common.SatoshisPerBTC
-	}
-	return balance, nil
+// ListWithdrawals lists the withdrawal history
+func (b *BTCMarkets) ListWithdrawals() ([]TransferData, error) {
```

Comment:
This seems to be missing the optional parameters:
https://api.btcmarkets.net/doc/v3#tag/Fund-Management-APIs/paths/~1v3~1withdrawals/get